packages/components/src/select-control/index.tsx

```diff
@@ -16,15 +18,39 @@ import { Icon, chevronDown } from '@wordpress/icons';
  */
 import BaseControl from '../base-control';
 import InputBase from '../input-control/input-base';
+import type { InputBaseProps, LabelPosition } from '../input-control/types';
 import { Select, DownArrowWrapper } from './styles/select-control-styles';
+import type { Size } from './types';
+import type { PolymorphicComponentProps } from '../ui/context';
 
-function useUniqueId( idProp ) {
+function useUniqueId( idProp?: string ) {
 	const instanceId = useInstanceId( SelectControl );
 	const id = `inspector-select-control-${ instanceId }`;
 
 	return idProp || id;
 }
 
+export interface SelectControlProps extends Omit< InputBaseProps, 'children' > {
+	help?: string;
+	hideLabelFromVision?: boolean;
+	multiple?: boolean;
+	onBlur?: ( event: FocusEvent< HTMLSelectElement > ) => void;
+	onFocus?: ( event: FocusEvent< HTMLSelectElement > ) => void;
+	onChange?: (
+		value: string | string[],
+		extra?: { event?: ChangeEvent< HTMLSelectElement > }
+	) => void;
+	options?: {
+		label: string;
+		value: string;
+		id?: string;
+		disabled?: boolean;
+	}[];
+	size?: Size;
+	value?: string | string[];
+	labelPosition?: LabelPosition;
+}
```

Comment:
Are all of these props supposed to be optional?

Reply:
They all have default values or default handling so yes, I think so. Did you notice any in particular that you think should not be optional?

Reply:
> Did you notice any in particular that you think should not be optional?

No, I didn't. It just felt unusual at a first glance, but as you said, all props seem to either be handled or passed to subcomponents for which they are also optional.